File: Karthik/14-9_to_15-9/minsubarraysum.py
class Solution(object):
    def minSubArrayLen(self, target, nums):
        """
        :type target: int
        :type nums: List[int]
        :rtype: int
        """
        i=0
        sumof=0
        minimum=1e9
        for j in range(len(nums)):
            sumof+=nums[j]
            while sumof>=target:
                minimum=min(minimum,j-i+1)
                sumof-=nums[i]
                i+=1
        if minimum==1e9:
            return 0
        return minimum
                    
        

File: Nithya/14-09-22/minsubarraysum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums)
    {
       int length=nums.size();
    int i=0;
        int j=0;
        int sum=0;
        int minimum=INT_MAX;
        while(j<length)
        {
            sum=sum+nums[j];
            if(sum>=target)
            {
                while(sum>=target)
                {
                    sum=sum-nums[i];
                    i++;
                }
                minimum=min(minimum,j-i+2);
            }
            j++;
        }
        return minimum==INT_MAX?0:minimum;
    }
};